### PR TITLE
fix(curriculum): step 13 incomplete instructions

### LINKED ISSUE #55790: dice game (js beta) step 13 incomplete instructions
### Describe the Issue

step 13 of the dice game (js beta) project asks the user to create a new function but neglects to mention that this function should expect an array argument.
(looking at the step 14, we can see the code there for the function is written with an array as an argument).
I believe the step will fail if the user attempts to use the global diceValuesArr global directly instead of defining the array argument and using it. 

Please adjust the step instructions to indicate the required parameter.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/review-algorithmic-thinking-by-building-a-dice-game/step-13

### Your code

N/A

### Expected behavior

if an array should be passed into the function detectFullHouse, then the instructions should say this explicitly. (it would be great also if the hints made that suggestion too)

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55790

### CODE REVIEW

**Comment by hbar1st** [comment]
will this commit also address updating the hints to detect if the array argument is missing and display an appropriate prompt?

**Comment by gagan-bhullar-tech** [comment]
@hbar1st added a hint to check if the function contains a argument.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ You should have a `detectFullHouse` function.
 assert.isFunction(detectFullHouse);
 ```
 
+Your `detectFullHouse` function should have a parameter.
+
+```js
+assert.match(code, /const\s+detectFullHouse\s*=\s*(\(\s*\w+\s*\))\s*=>\s*/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md:27]: This should work with a normal function declaration as well.

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md (hunk 1 above)
Comment (gagan-bhullar-tech) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md]: @lasjorg updated PR.

**Comment by gagan-bhullar-tech** [comment]
@lasjorg updated PR.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ You should have a `detectFullHouse` function.
 assert.isFunction(detectFullHouse);
 ```
 
+Your `detectFullHouse` function should have a parameter.
+
+```js
+assert.match(code, /((const\s+detectFullHouse\s*=\s*(\(\s*\w+\s*\))\s*=>\s*)|(function\s*detectFullHouse\(\s*\w+\s*\)))/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md:27]: The arrow function should allow `var` and `let`, and the parentheses should be optional when it is declared with just a single parameter.

The space between the `function` keyword and the name is not optional.

Something like this should work.

```suggestion
assert.match(code, /(?:var|let|const)\s+detectFullHouse\s*=\s*(\(\s*\w+\s*\)|\s*\w+\s*)\s*=>\s*|function\s+detectFullHouse\s*\(\s*\w+\s*\)/);
```

**Review by lasjorg — CHANGES_REQUESTED** [review]

**Comment by gagan-bhullar-tech** [comment]
@lasjorg thanks. Updated PR.

**Review by lasjorg — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**